applications/ShallowWaterApplication/custom_processes/depth_integration_process.cpp

```diff
@@ -159,6 +159,9 @@ void DepthIntegrationProcess<TDim>::Integrate(
             velocity = InterpolateVelocity(p_elem, rShapeFunctionsValues);
         }
     }
+    if (height < 1e-4) { // Sanity check to avoid Nan
```

Comment:
Not sure about the physics involved, but I'd use a smaller value for the zero check (of the order of 1.0e-12).